srtcore/list.cpp

```diff
@@ -865,3 +865,64 @@ srt::CRcvFreshLoss::Emod srt::CRcvFreshLoss::revoke(int32_t lo, int32_t hi)
 
     return DELETE;
 }
+
+bool srt::CRcvFreshLoss::removeOne(std::deque<CRcvFreshLoss>& w_container, int32_t sequence, int* pw_had_ttl)
+{
+    size_t i       = 0;
+    int    had_ttl = 0;
+    for (i = 0; i < w_container.size(); ++i)
+    {
+        had_ttl = w_container[i].ttl;
+        switch (w_container[i].revoke(sequence))
+        {
+        case CRcvFreshLoss::NONE:
+            continue; // Not found. Search again.
+
+        case CRcvFreshLoss::STRIPPED:
+            goto breakbreak; // Found and the modification is applied. We're done here.
+
+        case CRcvFreshLoss::DELETE:
+            // No more elements. Kill it.
+            w_container.erase(w_container.begin() + i);
+            // Every loss is unique. We're done here.
+            goto breakbreak;
+
+        case CRcvFreshLoss::SPLIT:
+            // Oh, this will be more complicated. This means that it was in between.
+            {
+                // So create a new element that will hold the upper part of the range,
+                // and this one modify to be the lower part of the range.
+
+                // Keep the current end-of-sequence value for the second element
+                int32_t next_end = w_container[i].seq[1];
+
+                // seq-1 set to the end of this element
+                w_container[i].seq[1] = CSeqNo::decseq(sequence);
+                // seq+1 set to the begin of the next element
+                int32_t next_begin = CSeqNo::incseq(sequence);
+
+                // Use position of the NEXT element because insertion happens BEFORE pointed element.
+                // Use the same TTL (will stay the same in the other one).
+                w_container.insert(w_container.begin() + i + 1,
+                                   CRcvFreshLoss(next_begin, next_end, w_container[i].ttl));
+            }
+            goto breakbreak;
+        }
+    }
+
+    // Could have made the "return" instruction instead of goto, but maybe there will be something
+    // to add in future, so keeping that.
+breakbreak:
```

Comment:
I would suggest to remove this `goto`,  now that it has been moved to a separate function.

Reply:
Breaking two loops at once is exactly this unique case of `goto` where it should be used. Alternatives are neither safer nor more efficient.

Reply:
I don't like this `goto` as well. Although it was in the initial source code moved here.

If `had_ttl` is just replaced by `*pw_had_ttl`, then all the `goto` statements can be replaced by `return`.
I believe it would be cleaner.